### DISCUSSION TITLE
ESP32CAM and TP5000 charger

### DISCUSSION THREAD

**Discussion by kapabel** [micropython/micropython#9981] [Hardware & Peripherals]
G'day,
I want to operate an ESP32Cam with a battery. I want to connect a LiFePO4 (32700) battery directly to the 3.3V of the ESP32Cam. But now I need to find a way to charge the battery.

For this I thought of a TP5000 module. My question is, can I connect this module directly?

According to my research, the ESP32 has at the 3.3V pin a maximum tolerance of 3.6V, so exactly what the TP5000 has as charging voltage.

Will I fry my ESP32 with this or do I have to have another protective mechanism in between?
Or is there a smarter solution out there?

Thanks!

**Comment by davefes** [comment]
The TP5000 specs on Amazon suggest 3.6/4.2V  I would not chance-it.  If you have a 5-6VDC source I would use a HZ7333 or a TO220 LM317.

**Reply by kapabel** [reply]
It can be either 3.6 or 4.2 depending if two solder points are soldered together or not. I would solder them together to get 3.6V.

**Reply by davefes** [reply]
And the tolerance on the 3.6V?

**Reply by elschopi** [reply]
To be safe, I'd put a diode with a low voltage drop in between the ESP32 input and the battery / charging module. ESP32 input voltage range is 2.2 to 3.6 V, so with a diode drop of 0.6 V you'd be fine. Smaller drop would of course be better ;) 
Only problem I see is supplying peripheral hardware with 3.3 V. I don't know voltage requirements of the camera module, but it could (should?) work...

**Reply by kapabel** [reply]
Good idea! Schottky diodes have a 0.2V Voltage drop. This could be what I need! Thanks!
